### PR TITLE
Created Children CRUD Endpoints

### PR DESCRIPTION
**Pull request by EduarddoEscobar**
Co-authored-by: Lizzy Thomson @lizzythomson 

## Description
Built out the ChildRouter to allow parents to CRUD their children.

New Endpoints: 

- Created a new POST endpoint at /children
- Created a new PUT endpoint at /children/:child_id
- Created a new DELETE endpoint at /children/:child_id

New Middleware:

- isChildParent: checks if the user making the request is the parent of the child from the request URL
- checkChildObject: No longer checks for a profile that was made before. Now checks for the information needed to create both the child and profile

New DB Access Functions: 

- addChild: Makes a profile for the child then creates the child
- updateChild: Updates a child with the given changes
- removeChild: Removes a child with the given id
- findChildParent: Finds the parent's profile id for the given child

Updated DB Access Functions: 

- getParentChildren: Now also returns profile information for the children

## Loom Video

https://www.loom.com/share/7276a782196e4812bd59ab8c0397b70d

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by joeys1122 — APPROVED** [review]
Great job getting these endpoints up Eduardo and Lizzy, your code is very clean and adds a ton of functionality!

**Review by dconley1212 — APPROVED** [review]
This looks great! Refactored the code base to look a lot better and you have the endpoints all working!

**Review by peterdavidconley — APPROVED** [review]
Middleware, Model.js, and README descriptions all look solid. Great job at cleaning up the console logs and other unnecessary code. Great use of the async/await for the ChildRouter post endpoint.

File: api/children/childrenMiddleware.js [hunk 1]
```diff
@@ -29,23 +30,44 @@ const isChildAlreadyEnrolled = async (req, res, next) => {
...
 };
 
+const isChildParent = async (req, res, next) => {
+  const { child_id } = req.params;
+  const { profile_id } = req.profile;
+  const [parent] = await Children.findChildParent(child_id);
+  if (parent.profile_id === profile_id) {
+    next();
+  } else {
+    next({ status: 403, message: 'This is not your child please stay away!' });
```
Comment (paulstgermain) [inline api/children/childrenMiddleware.js]: We _might_ want to adjust this error message to be slightly less accidentally hostile - keep in mind, we want to hand useful error messages up to the FE that don't necessarily accuse our users!

**Review by paulstgermain — CHANGES_REQUESTED** [review]
This is overall solid work - consistently structured, clean, and without introducing bugs! Just one change to make in our error handling and this would be good to merge up!

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**